code/modules/mob/dead/observer/observer.dm

```diff
@@ -31,6 +31,7 @@ var/list/image/ghost_images_simple = list() //this is a list of all ghost images
 	var/image/ghostimage_simple = null //this mob with the simple white ghost sprite
 	var/ghostvision = 1 //is the ghost able to see things humans can't?
 	var/seedarkness = 1
+	var/observetarget = null	//The target mob that the ghost is observing. Used as a reference in logout()
```

Comment:
type this as a mob, since it should only ever be a mob.
then remove the checks.